### PR TITLE
[bugfix] Correct the advisory rerun message on test failures

### LINKED ISSUE #1703: "Rerun with" advisory error message is a bit unclear
A NERSC staffer was a bit confused today when trying to run a particular ReFrame test and ReFrame responded with the error message:

```
[2021-01-21T16:48:22-08:00]   * Rerun with '-n <...> -p PrgEnv-intel --system <...>'
```

However, if one copies+pastes exactly what is in that error message, it fails again because the message does not recommend the `-r` flag. It was simple to fix, but seems like the error message should include the `-r` flag since it says `Rerun with ...`.

What do you think?

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #1703

### CODE REVIEW

File: reframe/frontend/statistics.py [hunk 1]
```diff
@@ -58,11 +58,10 @@ def retry_report(self):
                 if t.check.current_environ:
                     environ_name = t.check.current_environ.name
 
-                key = '%s:%s:%s' % (t.check.name, partition_name, environ_name)
                 # Overwrite entry from previous run if available
-                messages[key] = (
-                    '  * Test %s was retried %s time(s) and %s.' %
-                    (t.check.info(), run, 'failed' if t.failed else 'passed')
+                messages[f'{t.check.name}:{partition_name}:{environ_name}'] = (
```
Comment (vkarak) [inline reframe/frontend/statistics.py:62]: Use double quotes also here for consistency with the rest of the lines.

File: reframe/frontend/statistics.py [hunk 2]
```diff
@@ -276,16 +275,16 @@ def performance_report(self):
             if t.check.perfvalues.keys():
                 if t.check.name != previous_name:
                     report_body.append(line_width * '-')
-                    report_body.append('%s' % t.check.name)
+                    report_body.append(f'{t.check.name}')
```
Comment (vkarak) [inline reframe/frontend/statistics.py:278]:
```suggestion
                    report_body.append(t.check.name)
```

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — APPROVED** [review]
lgtm

**Review by victorusu — APPROVED** [review]
`lgtm`

**Comment by vkarak** [comment]
@jenkins-cscs retry dom

**Pull request merged**